tests/unit/handlers/test__helpers.py

```diff
@@ -106,61 +161,122 @@ def test_valid_context_header(self):
     @staticmethod
     def _call_fut():
         from google.cloud.logging_v2.handlers import _helpers
 
-        return _helpers.get_trace_id()
+        return _helpers.get_request_data()
 
     def _helper(self, django_return, flask_return):
         django_patch = mock.patch(
-            "google.cloud.logging_v2.handlers._helpers.get_trace_id_from_django",
+            "google.cloud.logging_v2.handlers._helpers.get_request_data_from_django",
             return_value=django_return,
         )
         flask_patch = mock.patch(
-            "google.cloud.logging_v2.handlers._helpers.get_trace_id_from_flask",
+            "google.cloud.logging_v2.handlers._helpers.get_request_data_from_flask",
             return_value=flask_return,
         )
 
         with django_patch as django_mock:
             with flask_patch as flask_mock:
-                trace_id = self._call_fut()
+                result = self._call_fut()
 
-        return django_mock, flask_mock, trace_id
+        return django_mock, flask_mock, result
 
     def test_from_django(self):
-        django_mock, flask_mock, trace_id = self._helper("test-django-trace-id", None)
-        self.assertEqual(trace_id, django_mock.return_value)
+        django_expected = (_DJANGO_HTTP_REQUEST, _DJANGO_TRACE_ID)
+        flask_expected = (None, None)
+        django_mock, flask_mock, output = self._helper(django_expected, flask_expected)
+        self.assertEqual(output, django_expected)
 
         django_mock.assert_called_once_with()
         flask_mock.assert_not_called()
 
     def test_from_flask(self):
-        django_mock, flask_mock, trace_id = self._helper(None, "test-flask-trace-id")
-        self.assertEqual(trace_id, flask_mock.return_value)
+        django_expected = (None, None)
+        flask_expected = (_FLASK_HTTP_REQUEST, _FLASK_TRACE_ID)
+
+        django_mock, flask_mock, output = self._helper(django_expected, flask_expected)
+        self.assertEqual(output, flask_expected)
 
         django_mock.assert_called_once_with()
         flask_mock.assert_called_once_with()
 
     def test_from_django_and_flask(self):
-        django_mock, flask_mock, trace_id = self._helper(
-            "test-django-trace-id", "test-flask-trace-id"
-        )
+        django_expected = (_DJANGO_HTTP_REQUEST, _DJANGO_TRACE_ID)
+        flask_expected = (_FLASK_HTTP_REQUEST, _FLASK_TRACE_ID)
+
+        django_mock, flask_mock, output = self._helper(django_expected, flask_expected)
+
         # Django wins.
-        self.assertEqual(trace_id, django_mock.return_value)
+        self.assertEqual(output, django_expected)
 
         django_mock.assert_called_once_with()
         flask_mock.assert_not_called()
 
-    def test_missing(self):
-        django_mock, flask_mock, trace_id = self._helper(None, None)
-        self.assertIsNone(trace_id)
+    def test_missing_http_request(self):
+        flask_expected = (None, _FLASK_TRACE_ID)
+        django_expected = (None, _DJANGO_TRACE_ID)
```

Comment:
perhaps, worth splitting this into two separate tests - for django and for flask

Reply:
This test specifically is focused on the integration function that grabs from either django or flask, the django and flask components are mocked out here